### PR TITLE
[3.10] bpo-46433: _PyType_GetModuleByDef: handle static types in MRO (GH-30696)

### PR DESCRIPTION
**Pull request by encukou**
(cherry picked from commit 0ef08530124c5ca13a9394f4ac18bee8e6c66409)

https://bugs.python.org/issue46433

### CODE REVIEW

**Review by vstinner — APPROVED** [review]
LGTM.

There was just a minor change between while and for in the main branch, but with this change, 3.10 and main use "for", so it's better!

File: Objects/typeobject.c [hunk 1]
```diff
@@ -3708,21 +3708,20 @@ _PyType_GetModuleByDef(PyTypeObject *type, struct PyModuleDef *def)
     assert(PyTuple_GET_SIZE(mro) >= 1);
 
     Py_ssize_t i = 0;
```
Comment (erlend-aasland) [inline Objects/typeobject.c]: You can remove this line, since the loop index is now inlined.

**Review by erlend-aasland — CHANGES_REQUESTED** [review]

File: Objects/typeobject.c (hunk 1 above)
Comment (vstinner) [reply Objects/typeobject.c]: Oh, I didn't see the warning: ``unused variable ‘i’ [-Wunused-variable]``.

**Review by erlend-aasland — APPROVED** [review]
Thanks, lgtm

**Comment by encukou** [comment]
Thanks for the review!

**Pull request merged**